index.src.html

```diff
@@ -750,6 +756,130 @@ <h3 id="try-delivery" algorithm>
 </section>
 
 <section>
+  <h2 id="observers">Reporting Observers</h2>
+  Each <a>environment settings object</a> has an associated list of
+  <dfn>registered reporting observers</dfn> ({{ReportingObserver}} objects).
+   
+  <h3 id=interface-reporting-observer>Interface {{ReportingObserver}}</h3>
+
+  <pre class="idl">
+[NoInterfaceObject]
+interface ReportBody {
+};
+
+[NoInterfaceObject]
+interface Report {
+    readonly attribute DOMString type;
+    readonly attribute DOMString url;
+    readonly attribute ReportBody? body;
+};
+
+[Constructor(ReportingObserverCallback callback)]
+interface ReportingObserver {
+    void observe(optional ReportingObserverOptions options);
+    void disconnect();
+};
+
+callback ReportingObserverCallback = void (sequence&lt;Report> reports, ReportingObserver observer);
+
+dictionary ReportingObserverOptions {
+  sequence&lt;DOMString> types;
+};
+  </pre>
+  A {{ReportingObserver}} object can be used to observe some types of
+  <a>reports</a> from JavaScript when they are created.
+  
+  Issue: How do we specify which reports are observable?  Network-layer reports
```

Comment:
In theory, that should be specified when the report is queued, as a default "off" parameter?

Reply:
I think network-only reports should be the exception.  So how about a default-off "network-only" parameter?